Saasu.API.Client/Framework/BaseProxy.cs

```diff
@@ -292,57 +301,48 @@ protected virtual System.Net.Http.HttpResponseMessage GetResponseMessage(string
-                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _bearerToken);
-            }
+            return GetResponseMessage<T>(requestUri, postData, ContentType.AsContentTypeString());
+        }
 
-            MediaTypeFormatter mediaFormatter;
+        private static Lazy<HttpClient> Client {get;} = new Lazy<HttpClient>(() =>
+        {
+            var client = new HttpClient();
+            client.DefaultRequestHeaders.ConnectionClose = true;
+            return client;
+        });
+        
 
-            if (ContentType == RequestContentType.ApplicationXml)
-            {
-                mediaFormatter = new System.Net.Http.Formatting.XmlMediaTypeFormatter();
+        protected virtual HttpResponseMessage GetResponseMessage<T>(string requestUri, T postData, string contentType)
+        {
+            var client = Client.Value;
 
-            }
-            else
+            using (var request = new HttpRequestMessage(OperationMethod, requestUri))
             {
-                mediaFormatter = new System.Net.Http.Formatting.JsonMediaTypeFormatter();
-            }
-
+                
+                if (!string.IsNullOrEmpty(contentType))
+                    request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue(contentType));
 
-            HttpResponseMessage responseMsg = null;
-            if (OperationMethod == HttpMethod.Get)
-            {
-                responseMsg = client.GetAsync(requestUri).Result;
-            }
-            else if (OperationMethod == HttpMethod.Delete && postData == null)
-            {
-                responseMsg = client.DeleteAsync(requestUri).Result;
-            }
-            else if (OperationMethod == HttpMethod.Head)
-            {
-                var rqstMsg = new HttpRequestMessage(HttpMethod.Head, requestUri);
-                responseMsg = client.SendAsync(rqstMsg).Result;
-            }
-            else
-            {
-                //Note: Need to explicitly specify the content type here otherwise this call fails.
-                if (OperationMethod == HttpMethod.Put)
+                if (AuthenticationMethod == AuthenticationType.OAuth && !string.IsNullOrEmpty(_bearerToken))
                 {
-                    responseMsg = client.PutAsync<T>(requestUri, postData, mediaFormatter).Result;
+                    request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _bearerToken);
                 }
-                else
+
+                if (postData != null)
                 {
-                    responseMsg = client.PostAsync<T>(requestUri, postData, mediaFormatter).Result;
+                    request.Content = new ObjectContent<T>(postData, ContentType == RequestContentType.ApplicationXml
+                        ? (MediaTypeFormatter) new XmlMediaTypeFormatter()
+                        : new JsonMediaTypeFormatter());
                 }
+
+                return client.SendAsync(request).Result;

```

Comment:
This is going to block the call and it won't be async.  These changes will only allow a single http request at a time, where before multiple would have been supported.

Reply:
No, I don't think so.  HttpClient is intended to be shared, and is reentrant and thread safe.  Unless I'm missing something.

See https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/